lib/ansible/modules/files/lineinfile.py

```diff
@@ -480,12 +480,17 @@ def main():
     regexp = params['regexp']
     line = params.get('line', None)
 
+    if regexp == '':
+        module.warn(
+            'The regular expression is empty, which will match every line in the file.'
```

Comment:
I think we need to say "empty string".  We also may want to give an alternative for people who actually do want to match every line in the file.

Reply:
What do you mean by "give an alternative"? This warning could be annoying to someone who does want this capability, and I don't want to encourage folks to disable warnings with noisy error messages.

Reply:
Ah, that's perfect! I see what you're saying.